### PR TITLE
Support audio preview

### PR DESCRIPTION
**Pull request by kiroj305**
- Added a basic audio player
- Pass in `NFT_Format` param through the recipe creation

### CODE REVIEW

File: lib/widgets/audio_widget.dart [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+              buffered: playbackEvent.bufferedPosition,
+              total: playbackEvent.duration,
+            ));
+    _init();
+  }
+
+  Future<void> _init() async {
+    try {
+      await _player.setUrl(widget.file.path);
+    } catch (e) {
```
Comment (mustafapylons) [inline lib/widgets/audio_widget.dart]: more meaningful variable than "e"

**Review by mustafapylons — COMMENTED** [review]
one small fix, otherwise clean code.

File: lib/screens/mint_screen.dart [hunk 2]
```diff
@@ -31,10 +32,13 @@ class MintScreen extends StatelessWidget {
               child: Column(
                 crossAxisAlignment: CrossAxisAlignment.start,
                 children: [
-                  if (provider.nftFormat.format == kImageText)
-                    ImageWidget(file: provider.file!),
-                  if (provider.nftFormat.format == kVideoText)
-                    VideoWidget(file: provider.file!),
+                  provider.nftFormat.format == kImageText
```
Comment (kjawadDeveloper2) [inline lib/screens/mint_screen.dart]: Use spread operator.

File: lib/widgets/audio_widget.dart [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+class _AudioWidgetState extends State<AudioWidget> {
+  late AudioPlayer _player;
+  late Stream<DurationState> _durationState;
+
+  @override
+  void initState() {
+    super.initState();
+    _player = AudioPlayer();
+    _durationState = Rx.combineLatest2<Duration, PlaybackEvent, DurationState>(
```
Comment (kjawadDeveloper2) [inline lib/widgets/audio_widget.dart:25]: Whats your intention of using Rx here?

File: pubspec.yaml [hunk 4]
```diff
@@ -67,6 +67,8 @@ dependencies:
   media_info: ^0.9.0
 
   video_player: ^2.2.18
+  just_audio: ^0.9.19
```
Comment (kjawadDeveloper2) [inline pubspec.yaml:70]: This audio package is very buggy on the ios side. Is there any other package.

**Review by kjawadDeveloper2 — CHANGES_REQUESTED** [review]
Kindly solve these issues.

File: pubspec.yaml (hunk 4 above)
Comment (kiroj305) [reply pubspec.yaml:70]: @kjawadDeveloper2 , can you recommend one for me?

Comment (kiroj305) [reply pubspec.yaml:70]: I found https://pub.dev/packages/audio_manager as the best audio plugin in flutter. Let me know your thought.

**Comment by kiroj305** [comment]
@mustafapylons , @kjawadDeveloper2 
Please review the fixes done.

**Review by mustafapylons — APPROVED** [review]
Looks good to me after the fix.

**Review by kjawadDeveloper2 — APPROVED** [review]
Looking good from my end.

**Pull request merged**